### ISSUE TITLE
crashes when opening external APK

### ISSUE THREAD

**Issue by diehard88** [MartinStyk/AndroidApkAnalyzer#32]
Happened with two different APK files.
Samsung Galaxy Note 8 / sm-g950w
Oreo
APK Analyzer 2.3.2

I'm available if more logs needed.

```
3811 V WindowManager: Relayout Window{dd70495 u0 sk.styk.martin.apkanalyzer/sk.styk.martin.apkanalyzer.ui.activity.MainActivity}: viewVisibility=4 req=1080x2220 WM.LayoutParams{(0,0)(fillxfill) sim=#20 ty=1 fl=#81810100 pfl=0x1020000 wanim=0x10302fd vsysui=0x710 needsMenuKey=2 colorMode=0 naviIconColor=0}
08-05 22:54:53.219  1496  1912 I InputDispatcher: Dropping event because there is no touchable window at (644, 1311).
08-05 22:54:53.220 11088 11088 D ViewRootImpl@87db7c6[MainActivity]: Relayout returned: old=[0,0][1080,2220] new=[0,0][1080,2220] result=0x1 surface={valid=false 0} changed=false
08-05 22:54:53.221  1496  1912 I InputDispatcher: Dropping event because there is no touchable window at (644, 1311).
08-05 22:54:53.224  1496  1912 I InputDispatcher: Dropping event because there is no touchable window at (644, 1309).
08-05 22:54:53.225  1496  2155 D GameManagerService: handleForegroundChange(). pkgName: sk.styk.martin.apkanalyzer, clsName: sk.styk.martin.apkanalyzer.ui.activity.MainActivity,FgActivityName:sk.styk.martin.apkanalyzer/.ui.activity.MainActivity
08-05 22:54:53.225  6243 16216 I TrayUsageStatesWatcher: noteResumeComponent : ComponentInfo{sk.styk.martin.apkanalyzer/sk.styk.martin.apkanalyzer.ui.activity.MainActivity}
08-05 22:54:53.225  1496  2155 D GameManagerService: sem_perfomance_mode: 0
08-05 22:54:53.225  1496  1968 D MARsPolicyManager: onPackageResumedFG pkgName = sk.styk.martin.apkanalyzer, userId = 0
08-05 22:54:53.225  1496  2155 D GameManagerService: notifyResumePause(). pkg: sk.styk.martin.apkanalyzer, type: 0, isMinimized: false, isTunableApp: true
08-05 22:54:53.225  1496  2155 D GameManagerService:   notifyResumePause(). changed mKillNotiCount: 0
08-05 22:54:53.225  1496  2155 D GameManagerService:   notifyResumePause(). before send intent to GameIntentService
08-05 22:54:53.226 11088 11088 D AndroidRuntime: Shutting down VM
--------- beginning of crash2
08-05 22:54:53.227 11088 11088 E AndroidRuntime: FATAL EXCEPTION: main
08-05 22:54:53.227 11088 11088 E AndroidRuntime: Process: sk.styk.martin.apkanalyzer, PID: 11088
08-05 22:54:53.227 11088 11088 E AndroidRuntime: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=66860, result=-1, data=Intent { dat=content://com.android.providers.downloads.documents/document/raw:/storage/emulated/0/Download/bugheist.apk flg=0x1 }} to activity {sk.styk.martin.apkanalyzer/sk.styk.martin.apkanalyzer.ui.activity.MainActivity}: java.lang.NumberFormatException: For input string: "raw:/storage/emulated/0/Download/bugheist.apk"
08-05 22:54:53.227 11088 11088 E AndroidRuntime: 	at android.app.ActivityThread.deliverResults(ActivityThread.java:4491)
08-05 22:54:53.227 11088 11088 E AndroidRuntime: 	at android.app.ActivityThread.handleSendResult(ActivityThread.java:4534)
08-05 22:54:53.227 11088 11088 E AndroidRuntime: 	at android.app.ActivityThread.-wrap20(Unknown Source:0)
08-05 22:54:53.227 11088 11088 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1752)
08-05 22:54:53.227 11088 11088 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:105)
08-05 22:54:53.227 11088 11088 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:164)
08-05 22:54:53.227 11088 11088 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6940)
08-05 22:54:53.227 11088 11088 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
08-05 22:54:53.227 11088 11088 E AndroidRuntime: 	at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
08-05 22:54:53.227 11088 11088 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
08-05 22:54:53.227 11088 11088 E AndroidRuntime: Caused by: java.lang.NumberFormatException: For input string: "raw:/storage/emulated/0/Download/bugheist.apk"
08-05 22:54:53.227 11088 11088 E AndroidRuntime: 	at java.lang.Long.parseLong(Long.java:590)
08-05 22:54:53.227 11088 11088 E AndroidRuntime: 	at java.lang.Long.valueOf(Long.java:804)
08-05 22:54:53.227 11088 11088 E AndroidRuntime: 	at sk.styk.martin.apkanalyzer.util.file.RealPathUtils.a(Unknown Source:200)
08-05 22:54:53.227 11088 11088 E AndroidRuntime: 	at sk.styk.martin.apkanalyzer.util.file.ApkFilePicker.a(Unknown Source:32)
08-05 22:54:53.227 11088 11088 E AndroidRuntime: 	at sk.styk.martin.apkanalyzer.ui.activity.applist.searchable.AppListFragment.onActivityResult(Unknown Source:37)
08-05 22:54:53.227 11088 11088 E AndroidRuntime: 	at android.support.v4.app.FragmentActivity.onActivityResult(Unknown Source:69)
08-05 22:54:53.227 11088 11088 E AndroidRuntime: 	at android.app.Activity.dispatchActivityResult(Activity.java:7547)
08-05 22:54:53.227 11088 11088 E AndroidRuntime: 	at android.app.ActivityThread.deliverResults(ActivityThread.java:4487)
08-05 22:54:53.227 11088 11088 E AndroidRuntime: 	... 9 more
08-05 22:54:53.227  1496  1912 I InputDispatcher: Dropping event because there is no touchable window at (644, 1308).
08-05 22:54:53.230  1496  1912 I InputDispatcher: Dropping event because there is no touchable window at (644, 1306).
08-05 22:54:53.230 10901 10901 D ViewRootImpl@906257[PickActivity]: MSG_WINDOW_FOCUS_CHANGED 0
08-05 22:54:53.231  2908  2908 I SemPersonaManager: SecureFolder personaId = 150
08-05 22:54:53.231  4085  4085 I Launcher: onWindowVisibilityChanged : false
08-05 22:54:53.232  1496  1912 I InputDispatcher: Dropping event because there is no touchable window at (644, 1305).
08-05 22:54:53.233  1496  2155 D GameManagerService:   notifyResumePause(). after send intent to GameIntentService
08-05 22:54:53.233  1496  1968 V WindowManager: Relayout Window{72f45e0 u0 com.sec.android.app.launcher/com.sec.android.app.launcher.activities.LauncherActivity}: viewVisibility=4 req=1080x2220 WM.LayoutParams{(0,0)(fillxfill) sim=#120 ty=1 fl=#81910d02 pfl=0x20000 fmt=-2 wanim=0x10302ea vsysui=0x700 needsMenuKey=2 colorMode=0 dimDuration=300 naviIconColor=0 sfl=0x40}
08-05 22:54:53.234  1496  7103 D GamePkgDataHelper: getGamePkgDataIncServer(). sk.styk.martin.apkanalyzer
08-05 22:54:53.235  1496  3893 D Debug   : !@DumpState : SHIP
08-05 22:54:53.235  1496  3893 D Debug   : !@DumpState : debug level:0x4f4c
08-05 22:54:53.235  6335 11129 D GOS:GameIntentService: onHandleIntent. begin
08-05 22:54:53.235  1496  3893 D Debug   : !@Dumpstate : Finally, system will skip dumpstate
08-05 22:54:53.235  1496  1912 I InputDispatcher: Dropping event because there is no touchable window at (644, 1304).
08-05 22:54:53.235  6335 11129 D GOS:GameIntentService: onHandleIntent. GameIntentService. type 0
08-05 22:54:53.235  1496  3893 W ActivityManager: crash : sk.styk.martin.apkanalyzer,0
08-05 22:54:53.236  2702  2732 D ForegroundUtils: could not check pending caller
08-05 22:54:53.237  6335 11129 D GOS:PkgData: PkgData(), pkgName: sk.styk.martin.apkanalyzer
08-05 22:54:53.237  6335 11129 D GOS:PkgData: setCustomFeatureFlag, pkgName: sk.styk.martin.apkanalyzer, old featureFlag: 346781844236309267, new featureFlag: 274882278167
08-05 22:54:53.237  6335 11129 D GOS:ServiceManager: onTunableNonGameResume. [sk.styk.martin.apkanalyzer] 
08-05 22:54:53.237  4085  4085 D ViewRootImpl@c948ffa[LauncherActivity]: Relayout returned: old=[0,0][1080,2220] new=[0,0][1080,2220] result=0x1 surface={valid=false 0} changed=false
08-05 22:54:53.238  1496  3893 W ActivityManager:   Force finishing activity sk.styk.martin.apkanalyzer/.ui.activity.MainActivity
08-05 22:54:53.238  1496  1912 I InputDispatcher: Dropping event because there is no touchable window at (644, 1302).
08-05 22:54:53.239  1496  3893 W MultiScreenManagerService: moveTaskBackToDisplayIfNeeded(): root activity or app is null, task=TaskRecord{70ec98fd0 #13396 A=sk.styk.martin.apkanalyzer U=0 StackId=1 sz=2}, rootActivity=null
08-05 22:54:53.239  1496  1496 D CodecSolution: MultiWindowEventListener::onMultiWindowFocusChanged(0)
08-05 22:54:53.239  1496  1496 D MdnieScenarioControlService: Listener Real Multi Window State : true
08-05 22:54:53.239  1496  1496 D CS_HdrController: MultiWindowEventListener::onFocusStackChanged(0)
08-05 22:54:53.240  6243 16216 I Tr

.............   2nd instance / opening another external APK / same crash

08-05 23:26:50.233 17202 17202 E AndroidRuntime: FATAL EXCEPTION: main
08-05 23:26:50.233 17202 17202 E AndroidRuntime: Process: sk.styk.martin.apkanalyzer, PID: 17202
08-05 23:26:50.233 17202 17202 E AndroidRuntime: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=66860, result=-1, data=Intent { dat=content://com.android.providers.downloads.documents/document/raw:/storage/emulated/0/Download/com.samsung.android.pluginrecents_1.3.01.62-130162000_minAPI26(nodpi)_apkmirror.com.apk flg=0x1 }} to activity {sk.styk.martin.apkanalyzer/sk.styk.martin.apkanalyzer.ui.activity.MainActivity}: java.lang.NumberFormatException: For input string: "raw:/storage/emulated/0/Download/com.samsung.android.pluginrecents_1.3.01.62-130162000_minAPI26(nodpi)_apkmirror.com.apk"
08-05 23:26:50.233 17202 17202 E AndroidRuntime: 	at android.app.ActivityThread.deliverResults(ActivityThread.java:4491)
08-05 23:26:50.233 17202 17202 E AndroidRuntime: 	at android.app.ActivityThread.handleSendResult(ActivityThread.java:4534)
08-05 23:26:50.233 17202 17202 E AndroidRuntime: 	at android.app.ActivityThread.-wrap20(Unknown Source:0)
08-05 23:26:50.233 17202 17202 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1752)
08-05 23:26:50.233 17202 17202 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:105)
08-05 23:26:50.233 17202 17202 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:164)
08-05 23:26:50.233 17202 17202 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6940)
08-05 23:26:50.233 17202 17202 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
08-05 23:26:50.233 17202 17202 E AndroidRuntime: 	at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
08-05 23:26:50.233 17202 17202 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
08-05 23:26:50.233 17202 17202 E AndroidRuntime: Caused by: java.lang.NumberFormatException: For input string: "raw:/storage/emulated/0/Download/com.samsung.android.pluginrecents_1.3.01.62-130162000_minAPI26(nodpi)_apkmirror.com.apk"
08-05 23:26:50.233 17202 17202 E AndroidRuntime: 	at java.lang.Long.parseLong(Long.java:590)
08-05 23:26:50.233 17202 17202 E AndroidRuntime: 	at java.lang.Long.valueOf(Long.java:804)
08-05 23:26:50.233 17202 17202 E AndroidRuntime: 	at sk.styk.martin.apkanalyzer.util.file.RealPathUtils.a(Unknown Source:200)
08-05 23:26:50.233 17202 17202 E AndroidRuntime: 	at sk.styk.martin.apkanalyzer.util.file.ApkFilePicker.a(Unknown Source:32)
08-05 23:26:50.233 17202 17202 E AndroidRuntime: 	at sk.styk.martin.apkanalyzer.ui.activity.applist.searchable.AppListFragment.onActivityResult(Unknown Source:37)
08-05 23:26:50.233 17202 17202 E AndroidRuntime: 	at android.support.v4.app.FragmentActivity.onActivityResult(Unknown Source:69)
08-05 23:26:50.233 17202 17202 E AndroidRuntime: 	at android.app.Activity.dispatchActivityResult(Activity.java:7547)
08-05 23:26:50.233 17202 17202 E AndroidRuntime: 	at android.app.ActivityThread.deliverResults(ActivityThread.java:4487)
08-05 23:26:50.233 17202 17202 E AndroidRuntime: 	... 9 more


```

**Comment by MartinStyk** [comment]
Hi this should be fixed in new 2.4.0 release rolling out now.

**Comment by MartinStyk** [comment]
I will close this, feel free to reopen if you hit this issue again with 2.4.0
